File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ETertiaryFixedDimLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ETertiaryFixedDimLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl38
            AppTheme.STANDARD -> R.color.cstl38
            AppTheme.BLUE -> R.color.cbl38
            AppTheme.GREEN -> R.color.cgl38
            AppTheme.RED -> R.color.crl38
            AppTheme.NOTHING -> R.color.cnol38
        }
    }
}

File: lib/layer2_core/light/e_tertiary_fixed_dim_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ETertiaryFixedDimLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl38;
      case 'standard': return cstl38;
      case 'blue': return cbl38;
      case 'green': return cgl38;
      case 'red': return crl38;
      case 'nothing': return cnol38;
      default: return cstl38;
    }
  }
}
